source/WesternStatesWater.WaDE.Managers.Api/Handlers/V2/OverlayFeatureItemGetRequestHandler.cs

```diff
@@ -0,0 +1,31 @@
+using System.Threading.Tasks;
+using WesternStatesWater.Shared.Exceptions;
+using WesternStatesWater.Shared.Resolver;
+using WesternStatesWater.WaDE.Accessors.Contracts.Api.V2.Requests;
+using WesternStatesWater.WaDE.Accessors.Contracts.Api.V2.Responses;
+using WesternStatesWater.WaDE.Contracts.Api.Requests.V2;
+using WesternStatesWater.WaDE.Contracts.Api.Responses.V2;
+using WesternStatesWater.WaDE.Engines.Contracts;
+using WesternStatesWater.WaDE.Engines.Contracts.Ogc.Requests;
+using WesternStatesWater.WaDE.Engines.Contracts.Ogc.Responses;
+using WesternStatesWater.WaDE.Managers.Mapping;
+
+namespace WesternStatesWater.WaDE.Managers.Api.Handlers.V2;
+
+public class OverlayFeatureItemGetRequestHandler(AccessorApi.IRegulatoryOverlayAccessor overlayAccessor, IFormattingEngine formattingEngine) : IRequestHandler<OverlayFeatureItemGetRequest, OverlayFeatureItemGetResponse>
+{
+    public async Task<OverlayFeatureItemGetResponse> Handle(OverlayFeatureItemGetRequest request)
+    {
+        var searchRequest = request.Map<OverlaySearchRequest>();
+        var searchResponse = await overlayAccessor.Search<OverlaySearchRequest, OverlaySearchResponse>(searchRequest);
+        
+        if (searchResponse.Overlays.Count != 1)
+        {
+            throw new WaDENotFoundException($"{nameof(OverlayFeatureItemGetRequestHandler)} found {searchResponse.Overlays.Count} sites for request {request.Id}.");
+        }
```

Comment:
I don't know if it makes sense to _throw_ here. I think the assumption here is that it's bad user input if the resource is not found, so we should return a 404, but it's not _exceptional_. The user screwed up, not us. I don't think we should program in a way for consumers of the API to trigger exceptions, as a rule of thumb, since they're a relatively expensive operation.

That said, I think what we can do here is return an error (FunctionBase will need updated like WestDAAT but you're probably aware of that)

```suggestion
        if (searchResponse.Overlays.Count != 1)
        {
            return new OverlayFeatureItemGetResponse
            {
                Error = new NotFoundError
                {
                    PublicMessage = $"{nameof(OverlayFeatureItemGetRequestHandler)} found {searchResponse.Overlays.Count} sites for request {request.Id}."
                }
            };
        }
```

thoughts?

Reply:
also note the logic would return NotFound if Count was 2 which would be more like an exception/programmer error...